skyportal/handlers/api/thumbnail.py

```diff
@@ -253,3 +273,345 @@ def delete(self, thumbnail_id):
+                int(self.get_query_argument("numPerPage", 100)),
+                1000,
+            )
+        except ValueError:
+            return self.error(
+                f'Cannot parse inputs pageNumber ({page_number}) '
+                f'or numPerPage ({num_per_page}) as an integers.'
+            )
+
+        good_like = f"%thumbnails{'/__' * required_depth}/%"
+        bad_like = f"%thumbnails{'/__' * (required_depth + 1)}/%"
+
+        num_moved = 0
+        with self.Session() as session:
+            stmt = sa.select(Thumbnail).where(Thumbnail.type.in_(types))
+            stmt = stmt.where(
+                sa.or_(
+                    ~Thumbnail.file_uri.like(good_like),
+                    Thumbnail.file_uri.like(bad_like),
+                )
+            )
+            stmt = stmt.offset((page_number - 1) * num_per_page)
+            stmt = stmt.limit(num_per_page)
+            thumbnails = session.execute(stmt).scalars().unique().all()
+            for t in thumbnails:
+                if t.file_uri is None:
+                    continue
+
+                if alert_available:
+                    # check file exists and non empty
+                    # if not, delete and repost thumbnail from alerts
+                    check_thumbnail_file(t, self.current_user.id, session)
+
+                hash = hashlib.sha256(t.obj_id.encode('utf-8')).hexdigest()
+                subfolders = []
+                for i in range(required_depth):
+                    subfolders.append(hash[i * 2 : (i + 1) * 2])
+                subfolders = '/'.join(subfolders)
+                path = (
+                    'thumbnails'.join(t.file_uri.split('thumbnails')[:-1])
+                    + 'thumbnails'
+                )
+                filename = os.path.basename(t.file_uri)
+                new_file_uri = os.path.join(path, subfolders, filename)
+                new_public_url = os.path.join(
+                    '/static/thumbnails', subfolders, filename
+                )
+                old_file_uri = t.file_uri
+
+                try:
+                    os.makedirs(os.path.dirname(new_file_uri), exist_ok=True)
+                    os.rename(old_file_uri, new_file_uri)
+                except Exception as e:
+                    return self.error(
+                        f'Could not move {old_file_uri} to {new_file_uri}: {e}'
+                    )
+
+                try:
+                    t.file_uri = new_file_uri
+                    t.public_url = new_public_url
+                    session.add(t)
+                    session.commit()
+                except Exception as e:
+                    session.rollback()
+                    # make sure to move the files back as well
+                    os.rename(new_file_uri, old_file_uri)
+                    return self.error(f'Could not update database row: {e}')
+
+                num_moved += 1
+
+            stmt = sa.select(Thumbnail).where(Thumbnail.type.in_(types))
```

Comment:
If this is the same syntax as the get, maybe it could be refactored to a single function?

Reply:
good idea
let me try that